File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    
    let vowels =  ['a','e','i','o','u', 'A', 'E', 'I', 'O', 'U'];
    let ixs = []; 
    let ss = s.split(''); 
    let vxs = [] ; 
    for(let i =0 ; i < s.length ; i++){
        if( vowels.indexOf(s[i]) > -1){
            ixs.push(i) ; 
            vxs.push(s[i]);
        }
    }
    vxs.reverse(); 
    for(let i=0; i < ixs.length ; i++){
        ss[ixs[i]] = vxs[i] ;
    }
    return ss.join('')


};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        
        s= list(s)
        vl = ['a','A', 'e','E', 'i','I', 'o', 'O' , 'u' ,'U' ]
        key = [] 
        for i ,v in enumerate(s):

            if v in vl:
                key.append(i)

        l , r = 0 ,len(key)-1

        while l< r:

              s[key[l]] , s[key[r]] = s[key[r]] , s[key[l]]
              l +=1 
              r -=1 

        return ''.join(s)
